File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numset = set(nums)
        longest = 0
        for n in numset:
            if (n-1) not in numset:
                length = 1
                while (n+length) in numset:
                    length += 1
                if length > longest:
                    longest = length
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int maxLen = 0;
        Set<Integer> set = new HashSet<>();
        for(int x : nums)
            set.add(x);
        
        for(int startElem : set)
        {
            if(!set.contains(startElem - 1))
            {
                int len = 1;
                while(set.contains(startElem + len))
                    len ++;
                maxLen = Math.max(maxLen, len);
            }
        }

        return maxLen;
    }
}